components/director/internal/domain/application/service.go

```diff
@@ -310,19 +310,21 @@ func (s *service) Delete(ctx context.Context, id string) error {
 		return err
 	}
 
-	runtimes, err := s.getRuntimeNamesForScenarios(ctx, appTenant, scenarios)
-	if err != nil {
-		return err
-	}
-
 	validScenarios := removeDefaultScenario(scenarios)
-	if len(validScenarios) > 0 && len(runtimes) > 0 {
-		application, err := s.appRepo.GetByID(ctx, appTenant, id)
+	if len(validScenarios) > 0 {
+		runtimes, err := s.getRuntimeNamesForScenarios(ctx, appTenant, validScenarios)
 		if err != nil {
-			return errors.Wrapf(err, "while getting application with id %s", id)
+			return err
+		}
+
+		if len(validScenarios) > 0 && len(runtimes) > 0 {
```

Comment:
minor: you've already checked for `len(validScenarios)` on 314 (can be omitted here)